libraries/lts/include/mcrl2/lts/detail/liblts_coupledsim.h

```diff
@@ -0,0 +1,592 @@
+    bool preserve_divergences = true;
+    weak_bisimulation_reduce(l1,preserve_divergences);
+    weak_bisimulation_reduce(l2,preserve_divergences);
+
+    mCRL2log(log::debug)
+      << "Input LTSs now reduced "
+      << "(preserve_divergences:" << (preserve_divergences ? "true" : "false") << "), "
+      << "continue with reduced version."
+      << std::endl;
+
+    std::set<cs_game_node> attacker_nodes;  // (flag=NODE_ATK, placeholder, node::int, node::int)
+    std::set<cs_game_node> defender_nodes;  // (flag, act::int, (node:int, node::int))
+
+    std::map<cs_game_node,std::set<cs_game_node>> predecessors;
+    std::map<cs_game_node,int> successor_count;
+    std::map<cs_game_node,int> node_winner;
+    const int WIN_DEFENDER = 0, WIN_ATTACKER = 1;
+
+    std::set<transition> l1_weak_transitions, l2_weak_transitions;
+
+    std::set<cs_game_move> moves;  // moves (node,node)
+    std::string move_label; // label as string representation.
+    std::ostringstream stream; // bypassing behavior (workaround for DEBUG)
+
+    /* Define game nodes here. */
+
+    /* Get Weak transitions. */
+    std::stack<transition> todo_weak;
+    // std::set<transition> l1_weak_transitions;
+    // std::set<transition> l2_weak_transitions; // do I need to save them?
+
+    /* filter transitions of t2. */
+    std::map<size_t, std::map<transition, bool>>  // if strong transition on true
+      l2_tran_from_node, l2_tran_into_node,
+      l1_tran_from_node, l1_tran_into_node;
+
+    mCRL2log(log::debug)
+      << "Find weak transitions."
+      << std::endl;
+
+    { // restructure l1 => get meta data and chain weak transitions.
+      for (const transition t1 : l1.get_transitions())
+      {
+        l1_tran_from_node[t1.from()][t1] = true;  // outgoing
+        l1_tran_into_node[t1.to()][t1] = true;  // incoming
+
+        /* Every transition is a weak transition, append to todo. */
+        todo_weak.push(transition(t1.from(), t1.label(), t1.to()));
+
+        l1_weak_transitions.insert(transition(t1.from(), t1.label(), t1.to()));
+
+        // add tau loop for everyone.
+        l1_weak_transitions.insert(transition(t1.from(), 0, t1.from()));
+        l1_weak_transitions.insert(transition(t1.to(), 0, t1.to()));
+      }
+
+      /* Add weak transititions. */
+      // on branching copy path and add all branches fins as fins.
+      while (!todo_weak.empty())
+      {
+        // pop and keep just start and extension.
+        // finish if next is second not tau.
+        transition weak = todo_weak.top();
+        todo_weak.pop();
+        size_t f = weak.from();
+        size_t l = weak.label();
+        size_t t = weak.to();
+        bool already_good = !l1.is_tau(l);  // path already has a good action
+
+        std::map<transition,bool> next = l1_tran_from_node[t];
+        size_t len = next.size();
+
+        if (true)  // also just tau chains may be later used
+          // if (already_good)  // (actually already) done
+        {
```

Comment:
Here, the comments indicate an alternative condition, but it is not clear what should be chosen. Same on line 285.

Reply:
With this whole block, starting at L200, I wanted to create weak transitions, since I wasn't sure there was a already method supporting such things. So I created my own `weak_transitions` struct there and fetched all weak transitions.

The actual `if (true)` block had an old condition, I didn't use anymore and was later abused as indentation.